syncer/syncer.go

```diff
@@ -1796,8 +1796,16 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 						s.isReplacingErr = true
 						// revert currentLocation to startLocation
 						currentLocation = startLocation
+					} else if op == pb.ErrorOp_Skip {
```

Comment:
Do we need to trigger a flush before update the checkpoint?

Reply:
I think all handle-error operation is for ddl. Before ddl excute, all dml is flushed.

Reply:
Now your logic is 
```
dmls
          <-- MatchAndApply, skip
ddl
         <-- saveGlobalPoint, flushCheckpoints
```
The previous dml are not flushed